libs/core/src/lib/utils/directives/truncate/truncate.directive.ts

```diff
@@ -0,0 +1,220 @@
+import {
+    AfterViewInit,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnChanges,
+    OnDestroy,
+    Output,
+    Renderer2,
+    SimpleChanges
+} from '@angular/core';
+import { fromEvent, Subscription } from 'rxjs';
+import { coerceNumberProperty } from '@angular/cdk/coercion';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+export enum TruncateCharCount {
+    DEFAULT = 500,
+    LESS = 300
+}
+
+@Directive({
+    // tslint:disable-next-line:directive-selector
+    selector: '[fd-truncate-target]',
+    exportAs: 'fdTruncateTarget'
+})
+export class TruncateTargetDirective implements OnChanges, AfterViewInit {
+    /**
+     * Target text for clamping
+     */
+    @Input()
+    fdTruncateTargetText: string;
+
+    /**
+     * Event with target instance for clamping
+     */
+    @Output() update = new EventEmitter<TruncateTargetDirective>();
+
+    /** @hidden */
+    constructor(private readonly _elementRef: ElementRef) {}
+
+    /**
+     * Native element of clamping target
+     */
+    get targetElement(): HTMLElement {
+        return this._elementRef.nativeElement;
+    }
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        this.update.emit(this);
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this.update.emit(this);
+    }
+}
+
+@Directive({
+    // tslint:disable-next-line:directive-selector
+    selector: '[fd-truncate]',
+    exportAs: 'fdTruncate'
+})
+export class TruncateDirective implements OnChanges, AfterViewInit, OnDestroy {
```

Comment:
I don't fully understand for what we need two directives, I think it can be done using only one

```
<span fdTruncate [fdTruncateText]="text" [fdTruncateChars]="chars" [fdTruncateState]="state"></span>
```